### ISSUE TITLE
Design the ux around in person non claim code badge issuance

### ISSUE THREAD

**Issue by threeqube** [mozilla/openbadger#78]
Reference issue: https://github.com/mozilla/CSOL-site/issues/92

As per a discussion with @threeqube @erinknight and @cmcavoy this is a situation where a learner would be assessed in person and then issued a digital badge in that same moment. We have been told that issuers will be given ipads or tablets specifically for this purpose.

We could design an admin login interface for csol that would give the user a dashboard view for issuing badges. Another option would be to do a spin off instance of open badger for CSOL.

Who gets admin access?
- We should have orgs give us lists of emails of those that are allowed to issue badges. 
- We'll use Persona for admins to log in.
- We'll identify who is admin and differentiate what they see

Suggested user flow
- in person, earner does some activity
- Instructor on the spot evaluates it
- goes to csol admin panel which is powered by Open Badger
- logs in
- see their orgs list of badges that are issuable
- selects badge to issue
- inputs learner email to issue that badge to the learner

**Comment by threeqube** [comment]
@iamjessklein, in a spontaneous discussion with @cmcavoy @erinknight and @carlacasilli today, we realized the following:
- Requesting tablets from the city, deploying them, training orgs not only how to use the tablets but also Open Badger admin, will be HARD. 
- Logistics behind printing tens of thousands of badges with unique claim codes for all the participating orgs will be HARD.
- There needs to be another option of helping orgs issue badges beyond claim codes and Open Badger. 

So @erinknight came up with the following idea: the userflow will be very similar to the above. 

When an instructor is ready to issue a badge, they fill out a form, online or paper, where they indicate which badge should be awarded to which students. They send that form over to us, online, mail or messager, we input that information and issue the badge to the students on behalf of the organization through Open Badger.

Let me know if you have any questions.

**Comment by eknight** [comment]
Just to be clear, I don't think this is an OR, but an AND. We need the CSOL issuing interface (mobile friendly), as well as so low level way they can send us info on the badges to issue for them.

**Comment by iamjessklein** [comment]
That works, in terms of design it's not too different from the original approach - so no work lost.

**Comment by iamjessklein** [comment]
![issuer](https://f.cloud.github.com/assets/535012/487854/346c92c8-b96b-11e2-928f-835a2f21589d.png)

![issuer_3](https://f.cloud.github.com/assets/535012/487856/38af78a0-b96b-11e2-92f5-1096ae38549e.png)

![mockup](https://f.cloud.github.com/assets/535012/487865/c994639e-b96b-11e2-9532-c7ce2f161d60.png)

![mockup_4](https://f.cloud.github.com/assets/535012/487899/2fb67ecc-b96d-11e2-884e-f7e83613dc64.png)

*\* MULTIPLE BADGEES **

![mockup_3](https://f.cloud.github.com/assets/535012/487896/ee85aa22-b96c-11e2-8aa4-921e77084615.png)

@brianloveswords let me know if we can do a csv file upload here, if not I will take it out of the mockups.

**Comment by threeqube** [comment]
I believe @cmcavoy said orgs should be able to upload csv files but will let him have the final word.

**Comment by iamjessklein** [comment]
After the mockups above - this is the workflow: 

![flow](https://f.cloud.github.com/assets/535012/497796/ff053efe-bc04-11e2-85f9-c8f9c6281ea8.png)
![flow](https://f.cloud.github.com/assets/535012/497811/7cead64e-bc05-11e2-9ab0-086a78e90c9c.png)
![flow_2](https://f.cloud.github.com/assets/535012/497821/ada2b536-bc05-11e2-99ff-35cbaf4dbcae.png)

I will mock out separately the workflow for <13

**Comment by cmcavoy** [comment]
:+1:

**Comment by cmcavoy** [comment]
Coming out of the training last week, it seems like this flow will be email based,
1. A mentor / issuer / org adds the learners email to the OpenBadger award badge interface.
2. The learner gets an email with a link to the badge acceptance. The link includes a claim code embedded in the url.
3. Learner clicks and lands on a badge acceptance page.
4. Learner logs in and accepts the badge.

**Comment by threeqube** [comment]
@iamjessklein do we need this page? 

![confirmation](https://f.cloud.github.com/assets/535012/487899/2fb67ecc-b96d-11e2-884e-f7e83613dc64.png)

Feels like we can just have text with the CTA [issue another badge] using same page as this one? 

![email-input](https://f.cloud.github.com/assets/535012/487865/c994639e-b96b-11e2-9532-c7ce2f161d60.png)

What do you think?

**Comment by threeqube** [comment]
Another question. We're not doing this email anymore right?

![email-learner](https://f.cloud.github.com/assets/535012/497796/ff053efe-bc04-11e2-85f9-c8f9c6281ea8.png)

Instead the user will have a notification indicator on their dashboard and when they go to the relevant badge, there will be some text there right?

**Comment by threeqube** [comment]
I don't think we need page either but let me know if you think otherwise @iamjessklein 

![confirmation-modal](https://f.cloud.github.com/assets/535012/497821/ada2b536-bc05-11e2-99ff-35cbaf4dbcae.png)

**Comment by cmcavoy** [comment]
We have this page already for bulk issuing, but we need a page to issue to
an individual who also gets evidence added to their badge. Use case is an
in person badge award where the issuer takes a picture of some work to
include, or includes a paragraph of text. We might be too far along to
actually implement this in the next 4 days, but...that's what was described
to us on the demo call last week.

On Mon, May 27, 2013 at 11:12 AM, threeqube notifications@github.comwrote:

> I don't think we need page either but let me know if you think otherwise
> @iamjessklein https://github.com/iamjessklein
> 
> [image: confirmation-modal]https://f.cloud.github.com/assets/535012/497821/ada2b536-bc05-11e2-99ff-35cbaf4dbcae.png
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/mozilla/openbadger/issues/78#issuecomment-18505535
> .

## 

@chmcavoy http://twitter.com/chmcavoy
http://lonelylion.com

**Comment by threeqube** [comment]
Wait, @cmcavoy which page are you referring to? I listed several in this ticket...

**Comment by cmcavoy** [comment]
@threeqube The page in OpenBadger where you can issue badges to a list of
email addresses.

On Mon, May 27, 2013 at 7:56 PM, threeqube notifications@github.com wrote:

> Wait, @cmcavoy https://github.com/cmcavoy which page are you referring
> to? I listed several in this ticket...
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/mozilla/openbadger/issues/78#issuecomment-18519648
> .

## 

@chmcavoy http://twitter.com/chmcavoy
http://lonelylion.com

**Comment by threeqube** [comment]
Got it. Thanks @cmcavoy
